### ISSUE TITLE
sat模型下载路径失效了

### ISSUE THREAD

**Issue by PaulX1029** [THUDM/VisualGLM-6B#261]
如题，微调所需要哦的sat的模型下载路径打不开了，visualglm-6b.zip文件可以去哪里获得呢

**Comment by 1049451037** [comment]
运行程序会自动下载

`pip install SwissArmyTransformer -U --no-deps`

直接运行这两行代码

```python
from model.visualglm import VisualGLMModel
model, args = VisualGLMModel.from_pretrained('visualglm-6b')
```

**Comment by PaulX1029** [comment]
@1049451037

**Comment by PaulX1029** [comment]
@1049451037 网络不稳定，有直接下载的地址吗？（捂脸）

**Comment by 1049451037** [comment]
没有，cloudflare对于大文件不支持直接链接。或者你可以使用rclone下载，相关配置在这里：https://github.com/THUDM/SwissArmyTransformer/blob/main/sat/resources/download.py#L79

**Comment by PaulX1029** [comment]
@1049451037 下载成功了大佬，但是我的机器在qlora微调时会被杀进程，内存32G，显存12G，报错如下：
[2023-09-04 14:01:16,174] [INFO] [RANK 0] building FineTuneVisualGLMModel model ...
/opt/miniconda3/envs/VisualGLM-6B/lib/python3.8/site-packages/torch/nn/init.py:405: UserWarning: Initializing zero-element tensors is a no-op
  warnings.warn("Initializing zero-element tensors is a no-op")
[2023-09-04 14:01:25,555] [INFO] [RANK 0] replacing layer 0 attention with lora
[2023-09-04 14:01:25,986] [INFO] [RANK 0] replacing layer 14 attention with lora
[2023-09-04 14:01:26,422] [INFO] [RANK 0] replacing chatglm linear layer with 4bit
[2023-09-04 14:01:45,719] [INFO] [launch.py:315:sigkill_handler] **Killing subprocess 2309229**
[2023-09-04 14:01:45,720] [ERROR] [launch.py:321:sigkill_handler] ['/opt/miniconda3/envs/VisualGLM-6B/bin/python', '-u', 'finetune_visualglm.py', '--local_rank=0', '--experiment-name', 'finetune-visualglm-6b', '--model-parallel-size', '1', '--mode', 'finetune', '--train-iters', '300', '--resume-dataloader', '--max_source_length', '64', '--max_target_length', '256', '--lora_rank', '10', '--layer_range', '0', '14', '--pre_seq_len', '4', '--train-data', './fewshot-data/dataset.json', '--valid-data', './fewshot-data/dataset.json', '--distributed-backend', 'nccl', '--lr-decay-style', 'cosine', '--warmup', '.02', '--checkpoint-activations', '--save-interval', '300', '--eval-interval', '10000', '--save', './checkpoints', '--split', '1', '--eval-iters', '10', '--eval-batch-size', '8', '--zero-stage', '1', '--lr', '0.0001', '--batch-size', '1', '--gradient-accumulation-steps', '4', '--skip-init', '--fp16', '--use_qlora'] exits with return code = -9

**Comment by 1049451037** [comment]
是不是你的内存还在被别的程序占用

**Comment by PaulX1029** [comment]
reboot过了，剩余29G是空闲的，所有程序全部关闭，只剩一个终端，能眼看着free到0，然后进程被杀
![image](https://github.com/THUDM/VisualGLM-6B/assets/77143808/25d8478c-d3c3-44ed-a290-f2360ca4041f)

**Comment by 1049451037** [comment]
那是不是你自己的dataset太大了，占用了很多内存，你是换了dataset吗

**Comment by PaulX1029** [comment]
@1049451037 用的您们提供的微调样例的dataset

**Comment by PaulX1029** [comment]
NCCL_DEBUG=info NCCL_IB_DISABLE=0 NCCL_NET_GDR_LEVEL=2 deepspeed --master_port 16666 --include localhost:0 --hostfile hostfile_single finetune_visualglm.py --experiment-name finetune-visualglm-6b --model-parallel-size 1 --mode finetune --train-iters 300 --resume-dataloader --max_source_length 64 --max_target_length 256 --lora_rank 10 --layer_range 0 14 --pre_seq_len 4 --train-data ./fewshot-data/dataset.json --valid-data ./fewshot-data/dataset.json --distributed-backend nccl --lr-decay-style cosine --warmup .02 --checkpoint-activations --save-interval 300 --eval-interval 10000 --save ./checkpoints --split 1 --eval-iters 10 --eval-batch-size 8 --zero-stage 1 --lr 0.0001 --batch-size 1 --gradient-accumulation-steps 4 --skip-init --fp16 --use_qlora
[2023-09-04 14:59:05,081] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-04 14:59:06,125] [WARNING] [runner.py:203:fetch_hostfile] Unable to find hostfile, will proceed with training with local resources only.
[2023-09-04 14:59:06,135] [INFO] [runner.py:570:main] cmd = /opt/miniconda3/envs/VisualGLM-6B/bin/python -u -m deepspeed.launcher.launch --world_info=eyJsb2NhbGhvc3QiOiBbMF19 --master_addr=127.0.0.1 --master_port=16666 --enable_each_rank_log=None finetune_visualglm.py --experiment-name finetune-visualglm-6b --model-parallel-size 1 --mode finetune --train-iters 300 --resume-dataloader --max_source_length 64 --max_target_length 256 --lora_rank 10 --layer_range 0 14 --pre_seq_len 4 --train-data ./fewshot-data/dataset.json --valid-data ./fewshot-data/dataset.json --distributed-backend nccl --lr-decay-style cosine --warmup .02 --checkpoint-activations --save-interval 300 --eval-interval 10000 --save ./checkpoints --split 1 --eval-iters 10 --eval-batch-size 8 --zero-stage 1 --lr 0.0001 --batch-size 1 --gradient-accumulation-steps 4 --skip-init --fp16 --use_qlora
[2023-09-04 14:59:06,949] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-04 14:59:07,823] [INFO] [launch.py:138:main] 0 NCCL_IB_DISABLE=0
[2023-09-04 14:59:07,824] [INFO] [launch.py:138:main] 0 NCCL_DEBUG=info
[2023-09-04 14:59:07,824] [INFO] [launch.py:138:main] 0 NCCL_NET_GDR_LEVEL=2
[2023-09-04 14:59:07,824] [INFO] [launch.py:145:main] WORLD INFO DICT: {'localhost': [0]}
[2023-09-04 14:59:07,824] [INFO] [launch.py:151:main] nnodes=1, num_local_procs=1, node_rank=0
[2023-09-04 14:59:07,824] [INFO] [launch.py:162:main] global_rank_mapping=defaultdict(<class 'list'>, {'localhost': [0]})
[2023-09-04 14:59:07,824] [INFO] [launch.py:163:main] dist_world_size=1
[2023-09-04 14:59:07,824] [INFO] [launch.py:165:main] Setting CUDA_VISIBLE_DEVICES=0
[2023-09-04 14:59:08,834] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes

 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
bin /opt/miniconda3/envs/VisualGLM-6B/lib/python3.8/site-packages/bitsandbytes/libbitsandbytes_cuda116.so
/opt/miniconda3/envs/VisualGLM-6B/lib/python3.8/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: /opt/miniconda3/envs/VisualGLM-6B did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] as expected! Searching further paths...
  warn(msg)
CUDA SETUP: CUDA runtime path found: /usr/local/cuda-11.6/lib64/libcudart.so
CUDA SETUP: Highest compute capability among GPUs detected: 8.6
CUDA SETUP: Detected CUDA version 116
CUDA SETUP: Loading binary /opt/miniconda3/envs/VisualGLM-6B/lib/python3.8/site-packages/bitsandbytes/libbitsandbytes_cuda116.so...
[2023-09-04 14:59:11,241] [INFO] using world size: 1 and model-parallel size: 1 
[2023-09-04 14:59:11,241] [INFO] > padded vocab (size: 100) with 28 dummy tokens (new size: 128)
[2023-09-04 14:59:11,243] [INFO] [RANK 0] > initializing model parallel with size 1
[2023-09-04 14:59:11,244] [INFO] [comm.py:637:init_distributed] cdb=None
[2023-09-04 14:59:11,244] [WARNING] [config_utils.py:69:_process_deprecated_field] Config parameter cpu_offload is deprecated use offload_optimizer instead
[2023-09-04 14:59:11,245] [INFO] [checkpointing.py:1030:_configure_using_config_file] {'partition_activations': False, 'contiguous_memory_optimization': False, 'cpu_checkpointing': False, 'number_checkpoints': None, 'synchronize_checkpoint_boundary': False, 'profile': False}
[2023-09-04 14:59:11,245] [INFO] [checkpointing.py:227:model_parallel_cuda_manual_seed] > initializing model parallel cuda seeds on global rank 0, model parallel rank 0, and data parallel rank 0 with model parallel seed: 3952 and data parallel seed: 1234
[2023-09-04 14:59:11,263] [INFO] [RANK 0] building FineTuneVisualGLMModel model ...
/opt/miniconda3/envs/VisualGLM-6B/lib/python3.8/site-packages/torch/nn/init.py:405: UserWarning: Initializing zero-element tensors is a no-op
  warnings.warn("Initializing zero-element tensors is a no-op")
[2023-09-04 14:59:20,580] [INFO] [RANK 0] replacing layer 0 attention with lora
[2023-09-04 14:59:21,013] [INFO] [RANK 0] replacing layer 14 attention with lora
[2023-09-04 14:59:21,445] [INFO] [RANK 0] replacing chatglm linear layer with 4bit
[2023-09-04 14:59:56,957] [INFO] [RANK 0]  > number of parameters on model parallel rank 0: 7802848768
[2023-09-04 15:00:00,923] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 126856

**Comment by PaulX1029** [comment]
@1049451037 这是完整的报错信息，能否看出一些其他的端倪呢？

**Comment by seyosum** [comment]
> 

大佬，权重文件能给个网盘链接不，网络老断，根本下载不了😭

**Comment by seyosum** [comment]
@PaulX1029

**Comment by PaulX1029** [comment]
@seyosum 还需要吗？

**Comment by hjx529126** [comment]
> @seyosum 还需要吗？
大佬，需要ya

**Comment by hjx529126** [comment]
@PaulX1029   大佬权重文件能给个网盘链接吗？？ 跪求了
